include/contraction/contractionGraph.hpp

```diff
@@ -70,6 +73,46 @@ class contractionGraph :
         return --min_edge_id;
     }
 
+    /*!
+        @brief for C calls: to get the metric of a node, directly from the graph
+        @param [in] v vertex_descriptor
+        @return int64_t: the value of the metric for node v
+    */
+    double get_M(int64_t vertex_id) {
+        return (this->graph[this->vertices_map[vertex_id]]).get_metric();
+    }
+
+    /*!
+        @brief for C calls: to get the order of a node, directly from the graph
+        @param [in] v vertex_descriptor
+        @return int64_t: the order of node v
+    */
+    int64_t get_O(int64_t vertex_id) {
+        return (this->graph[this->vertices_map[vertex_id]]).get_vertex_order();
+    }
+
+    /*!
+        @brief defines the metric and hierarchy at the level of the nodes, from a given priority queue
+        @param [in] PQ priority_queue
+        @return void
+    */
+    void set_vertices_metric_and_hierarchy(
+        PQ priority_queue,
+        std::ostringstream &log
+    ) {
+        int64_t i = 0;
+        while (!priority_queue.empty()) {
+            i++;
+            std::pair< double, V > ordered_vertex = priority_queue.top();
+            priority_queue.pop();
+            (this->graph[ordered_vertex.second]).add_contracted_vertex(this->graph[ordered_vertex.second]);
```

Comment:
Kind of hard to understand what is `this->graph[ordered_vertex.second]`, but it seems you add it as a contracted vertex to itself... Right?

Reply:
It was a first choice to return the shortcuts + all nodes of the graph, considered as contracted with themself (so that they appear in the result), but I agree it is useless afterwards. I remove it, so only the shortcuts will be returned by the query.